src/main/scala/com/microsoft/partnercatalyst/fortis/spark/ProjectFortis.scala

```diff
@@ -61,8 +63,13 @@ object ProjectFortis extends App {
       .set("spark.kryoserializer.buffer.max", "64m")
     CassandraConfig.init(conf, batchDuration, fortisSettings)
 
-    val sparksession = SparkSession.builder().config(conf).getOrCreate()
-    val ssc = new StreamingContext(sparksession.sparkContext, batchDuration)
+    val sparkContext = SparkContext.getOrCreate(conf)
+    val ssc = new StreamingContext(sparkContext, batchDuration)
+    ssc.checkpoint(fortisSettings.progressDir)
+    ssc
+  }
+
+  private def attachToContext(ssc:StreamingContext): Boolean = {
```

Comment:
I factored out `attachToContext` from `createStreamingContext` in order to call it in the loop below.